test/dutchExchange-Tulip.js

```diff
@@ -662,16 +667,427 @@ const c2 = () => contract('DutchExchange --> Tulip Flow --> 1 Seller + 2 Buyers
+    `)
+    assert.equal(sellVolumes, sellingAmount - svFee(0.5), 'sellVolumes === seller1Balance')
+  })
+  
+  it('BUYER1: Non Auction clearing PostBuyOrder + Claim => Tulips = 0', async () => {
+    eventWatcher(dx, 'ClaimBuyerFunds', {})
+    eventWatcher(dx, 'LogNumber', {})
+    log(`
+    ============================================================================================
+    T2.5: Buyer1 PostBuyOrder => [[Non Auction clearing PostBuyOrder + Claim]] => [[Tulips = 0]]
+    ============================================================================================
+    `)
+    log(`
+    BUYER1 GNO BALANCE = ${(await getBalance(buyer1, gno)).toEth()}
+    BUYER1 ETH BALANCE = ${(await getBalance(buyer1, eth)).toEth()}
+    `)
+    /*
+     * SUB TEST 1: MOVE TIME AFTER SCHEDULED AUCTION START TIME && ASSERT AUCTION-START =TRUE
+     */
+    await setAndCheckAuctionStarted(eth, gno)    
+    
+    // Should be 0 here as aucIdx = 1 ==> we set aucIdx in this case
+    const [closingNum, closingDen] = (await dx.closingPrices.call(eth.address, gno.address, 1))
+    // Should be 4 here as closing price starts @ 2 and we times by 2
+    const [num, den] = (await dx.getPriceForJS(eth.address, gno.address, 1)).map(i => i.toNumber())
+    log(`
+    Last Closing Prices:
+    closeN        = ${closingNum}
+    closeD        = ${closingDen}
+    closingPrice  = ${closingNum / closingDen}
+    ===========================================
+    Current Prices:
+    n             = ${num}
+    d             = ${den}
+    price         = ${num / den}
+    `)
+
+    /*
+     * SUB TEST 2: postBuyOrder => 20 GNO @ 4:1 price
+     * post buy order @ price 4:1 aka 1 GNO => 1/4 ETH && 1 ETH => 4 GNO
+     * @{return} ... 20GNO * 1/4 => 5 ETHER
+     */
+    await postBuyOrder(eth, gno, false, (20).toWei(), buyer1)
+    log(`
+    Buy Volume AFTER = ${((await dx.buyVolumes.call(eth.address, gno.address)).toNumber()).toEth()}
+    Left to clear auction = ${((await dx.sellVolumesCurrent.call(eth.address, gno.address)).toNumber() - ((await dx.buyVolumes.call(eth.address, gno.address)).toNumber()) * (den / num)).toEth()}
+    `)
+    let idx = await getAuctionIndex()
+    const [claimedFunds, tulipsIssued] = (await dx.claimBuyerFunds.call(eth.address, gno.address, buyer1, idx)).map(i => i.toNumber())
+    await checkUserReceivesTulipTokens(eth, gno, buyer1, idx)
+    // await dx.claimBuyerFunds(eth.address, gno.address, buyer1, idx)
+    // const { args, args: { returned: claimedFunds, tulipsIssued } } = dx.ClaimBuyerFunds((err, resp) => resp)
+    log(`
+    CLAIMED FUNDS => ${claimedFunds.toEth()}
+    TULIPS ISSUED => ${tulipsIssued.toEth()}
+    `)
+
+    assert.equal(tulipsIssued, 0, 'Tulips only issued / minted after auction Close so here = 0')
+  })
+
+  it(
+    'BUYER1: Tries to lock and unlock Tulips --> Auction NOT cleared --> asserts 0 Tulips minted and in mapping', 
+    async () => unlockTulipTokens(buyer1),
```

Comment:
unnecessary `async`, favorite keyword?

Reply:
no, just tired and not thinking properly, also yes

- will fix these in #160